WhisperAPI/WhisperAPI/Controllers/SuggestionsController.cs

```diff
@@ -31,8 +31,14 @@ public SuggestionsController(ISuggestionsService suggestionsService, IQuestionsS
         [HttpPost]
         public IActionResult GetSuggestions([FromBody] SearchQuery searchQuery)
         {
-            this._nlpService.UpdateAndAnalyseSearchQuery(searchQuery);
-            this._suggestionsService.UpdateContextWithNewQuery(this.ConversationContext, searchQuery);
+            var nlpAnalysis = this._nlpService.AnalyseSearchQuery(searchQuery);
+            ContextItem contextItem = new ContextItem
+            {
+                NlpAnalysis = nlpAnalysis,
+                SearchQuery = searchQuery,
+            };
+
+            this._suggestionsService.UpdateContextWithNewItem(this.ConversationContext, contextItem);
```

Comment:
Instead of doing your object in controller, you should pass nlpAnalysis and searchquery to this method. It's her job to create the correct object and to put it in context